graylog2-web-interface/src/components/indices/IndexSetFieldTypes/hooks/useCustomColumnRenderers.tsx

```diff
@@ -0,0 +1,83 @@
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import React, { useMemo } from 'react';
+import keyBy from 'lodash/keyBy';
+
+import type { FieldTypeOrigin } from 'components/indices/IndexSetFieldTypes/types';
+import ExpandedRowToggleWrapper from 'components/indices/IndexSetFieldTypes/originBadges/ExpandedRowToggleWrapper';
+import ProfileBadge from 'components/indices/IndexSetFieldTypes/originBadges/ProfileBadge';
+import OverriddenIndexBadge from 'components/indices/IndexSetFieldTypes/originBadges/OverriddenIndexBadge';
+import OverriddenProfileBadge from 'components/indices/IndexSetFieldTypes/originBadges/OverriddenProfileBadge';
+import IndexBadge from 'components/indices/IndexSetFieldTypes/originBadges/IndexBadge';
+import { Icon } from 'components/common';
+import useFieldTypesForMappings from 'views/logic/fieldactions/ChangeFieldType/hooks/useFieldTypesForMappings';
+import type { Attribute } from 'stores/PaginationTypes';
+
+const useCustomColumnRenderers = (attributes: Array<Attribute>) => {
+  const { data: { fieldTypes } } = useFieldTypesForMappings();
+  const normalizedOrigin = useMemo(() => {
+    const originOptions = attributes?.find(({ id }) => id === 'origin')?.filter_options;
+
+    return keyBy(originOptions, 'value');
+  }, [attributes]);
+
+  return useMemo(() => ({
+    attributes: {
+      type: {
+        renderCell: (item: string) => <span>{fieldTypes[item]}</span>,
+      },
+      origin: {
+        renderCell: (origin: FieldTypeOrigin, { id }) => {
+          switch (origin) {
+            case 'PROFILE':
+              return (
+                <ExpandedRowToggleWrapper id={id}>
+                  <ProfileBadge title={normalizedOrigin?.PROFILE?.title} />
+                </ExpandedRowToggleWrapper>
+              );
+            case 'OVERRIDDEN_INDEX':
+              return (
+                <ExpandedRowToggleWrapper id={id}>
+                  <OverriddenIndexBadge title={normalizedOrigin?.OVERRIDDEN_INDEX?.title} />
+                </ExpandedRowToggleWrapper>
+              );
+            case 'OVERRIDDEN_PROFILE':
+              return (
+                <ExpandedRowToggleWrapper id={id}>
+                  <OverriddenProfileBadge title={normalizedOrigin?.OVERRIDDEN_PROFILE?.title} />
+                </ExpandedRowToggleWrapper>
+              );
+            default:
+              return (
+                <ExpandedRowToggleWrapper id={id}>
+                  <IndexBadge title={normalizedOrigin?.INDEX?.title} />
```

Comment:
Since all components contain just a badge, wouldn't it be simpler to render the `Badge` in one component which maps the provided type with the `bsStyle`?

- In case we want to adjust the badge we would just have to do it in one place  
- this way we could get rid of four components.
- it would simplify this logic here, since we do not need a switch case anymore.
- here we just have to render one component and do not have to repeat `<ExpandedRowToggleWrapper id={id}>` etc.
- we can reuse the component in `OriginFilterValueRenderer.tsx` and do not need the object in `OriginFilterValueRenderer.tsx` which also maps the origin with these component, like in this switch case.

Reply:
Usually when we create an expandable section it offers some functionality. In this case we just display text. Wouldn't it be enough (and consistent) to use the `HoverForHelp` component / icon inside the badge, which includes the explanation?